packages/core/src/errors.ts

```diff
@@ -1,58 +1,176 @@
+import { ERRORS, ErrorDescriptor, getErrorCode } from "./errors-list";
+
 /**
- * All exceptions intenionally thrown with Ignition-core
- * extend this class.
+ * Base error class extended by all custom errors.
+ * Placeholder to allow us to customize error output formatting in the future.
  *
  * @alpha
  */
-export class IgnitionError extends Error {
+export class CustomError extends Error {
```

Comment:
Hardhat does a lot of clever stuff to modify the displayed output of errors inside this class. We currently aren't doing those things, but I wanted to leave room for us to add it later if we decide to.